lua/neotest-vstest/init.lua

```diff
@@ -92,17 +109,33 @@ function DotnetNeotestAdapter.is_test_file(file_path)
     logger.debug(
       "neotest-vstest: marking file as non-test file since no client was found: " .. file_path
     )
+    is_test_dir_cache[file_path] = false
     return false
   end
 
+  is_test_dir_cache[file_path] = true
   return true
 end
 
-function DotnetNeotestAdapter.filter_dir(name)
+function DotnetNeotestAdapter.filter_dir(name, rel_path, root)
   if name == "bin" or name == "obj" then
     return false
   end
 
+  -- Filter out directories that are not part of the solution (if there is a solution)
+  local fullpath = vim.fs.joinpath(root, rel_path)
```

Comment:
I'm pretty sure that the `name` parameter is already supposed to be the fully expanded path